url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/syntax/builder/SyntaxBuilderParserTables.java

```diff
@@ -0,0 +1,720 @@
+		s.addRightSymbol("`identifier`");
+		s.addRightSymbol("\"::=\"");
+		s.addRightSymbol("unionseq");
+		s.addRightSymbol("';'");
+
+
+		loadGotoTable();
+		loadParseActionTable();
+
+		terminalsWithoutEpsilon = new ArrayList(12);
+		terminalsWithoutEpsilon.add("`bnf_chardef`");
+		terminalsWithoutEpsilon.add("\"..\"");
+		terminalsWithoutEpsilon.add("`identifier`");
+		terminalsWithoutEpsilon.add("`ruleref`");
+		terminalsWithoutEpsilon.add("`stringdef`");
+		terminalsWithoutEpsilon.add("'-'");
+		terminalsWithoutEpsilon.add("'('");
+		terminalsWithoutEpsilon.add("')'");
+		terminalsWithoutEpsilon.add("`quantifier`");
+		terminalsWithoutEpsilon.add("'|'");
+		terminalsWithoutEpsilon.add("\"::=\"");
+		terminalsWithoutEpsilon.add("';'");
+	}
+	private void loadGotoTable()	{
+		gotoTable = new ArrayList(35);
+
+		loadGoto_0();
+		loadGoto_1();
+		gotoTable.add(null);  // state 2
+		loadGoto_3();
+		loadGoto_4();
+		gotoTable.add(null);  // state 5
+		loadGoto_6();
+		loadGoto_7();
+		gotoTable.add(null);  // state 8
+		loadGoto_9();
+		loadGoto_10();
+		gotoTable.add(null);  // state 11
+		gotoTable.add(null);  // state 12
+		loadGoto_13();
+		gotoTable.add(null);  // state 14
+		loadGoto_15();
+		gotoTable.add(null);  // state 16
+		gotoTable.add(null);  // state 17
+		gotoTable.add(null);  // state 18
+		gotoTable.add(null);  // state 19
+		gotoTable.add(null);  // state 20
+		loadGoto_21();
+		loadGoto_22();
+		gotoTable.add(null);  // state 23
+		gotoTable.add(null);  // state 24
+		loadGoto_25();
+		loadGoto_26();
+		gotoTable.add(null);  // state 27
+		loadGoto_28();
+		gotoTable.add(null);  // state 29
+		gotoTable.add(null);  // state 30
+		gotoTable.add(null);  // state 31
+		gotoTable.add(null);  // state 32
+		gotoTable.add(null);  // state 33
+		gotoTable.add(null);  // state 34
+	}
+	private void loadGoto_0()	{
+		Hashtable g = new Hashtable(3, 1);
+		gotoTable.add(g);
+		g.put("syntax", new Integer(3));
+		g.put("`identifier`", new Integer(1));
+		g.put("rule", new Integer(2));
+	}
+	private void loadGoto_1()	{
+		Hashtable g = new Hashtable(1, 1);
+		gotoTable.add(g);
+		g.put("\"::=\"", new Integer(4));
+	}
+	private void loadGoto_3()	{
+		Hashtable g = new Hashtable(2, 1);
+		gotoTable.add(g);
+		g.put("`identifier`", new Integer(1));
+		g.put("rule", new Integer(5));
+	}
+	private void loadGoto_4()	{
+		Hashtable g = new Hashtable(14, 1);
+		gotoTable.add(g);
+		g.put("`identifier`", new Integer(16));
+		g.put("'('", new Integer(15));
+		g.put("`ruleref`", new Integer(11));
+		g.put("intersectionunit", new Integer(18));
+		g.put("sequence_opt", new Integer(12));
+		g.put("unionseq", new Integer(6));
+		g.put("sequence", new Integer(9));
+		g.put("intersection", new Integer(8));
+		g.put("intersectionstartunit", new Integer(13));
+		g.put("`stringdef`", new Integer(17));
+		g.put("`bnf_chardef`", new Integer(7));
+		g.put("sequnit", new Integer(10));
+		g.put("set", new Integer(19));
+		g.put("quantifiedsequnit", new Integer(14));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 187:202, 243:258.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.